### PR TITLE
Feature/df 1842 search by beruf chisco support

### PR DESCRIPTION
**Pull request by ganqqwerty**
In ch fiche search we search by bfs and chisco, using reference-service-2 therefore **there are categories**: 
image
In fiche creation/edition we search only by bfs code, using reference-service-2 so there are **no categories**: 
image
In portal the job occupations are still being fetched from referenceservvice-1 
image
image

### CODE REVIEW

File: alv-portal-ui/src/app/competence-catalog/shared/occupation-label-with-ch-isco-repository.service.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { Injectable } from '@angular/core';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class OccupationLabelWithChIscoRepositoryService {
```
Comment (emina-str) [inline alv-portal-ui/src/app/competence-catalog/shared/occupation-label-with-ch-isco-repository.service.ts]: @ganqqwerty  we don't use this class, I think

File: alv-portal-ui/src/app/competence-catalog/ch-fiches/ch-fiches-overview/ch-fiches-overview.component.ts [hunk 2]
```diff
@@ -49,7 +50,7 @@ export class ChFichesOverviewComponent extends OverviewComponent<ChFiche> implem
   }
 
   loadOccupations(query: string): Observable<OccupationTypeaheadItem[]> {
-    return this.occupationSuggestionService.fetchCompetenceCatalogOccupations(query);
+    return this.occupationSuggestionService.fetchCompetenceCatalogOccupations(query, [OccupationTypes.BFS, OccupationTypes.CHISCO5]);
```
Comment (fabiangotzen) [inline alv-portal-ui/src/app/competence-catalog/ch-fiches/ch-fiches-overview/ch-fiches-overview.component.ts]: Why only CHISCO5? I think CHISCO3 is also needed here.

Comment (cjpsycos) [reply alv-portal-ui/src/app/competence-catalog/ch-fiches/ch-fiches-overview/ch-fiches-overview.component.ts]: No more for KK, only BFS and CHISCO5

File: alv-portal-ui/src/app/competence-catalog/shared/occupation-label-with-ch-isco-repository.service.ts (hunk 1 above)
Comment (ganqqwerty) [reply alv-portal-ui/src/app/competence-catalog/shared/occupation-label-with-ch-isco-repository.service.ts]: thnaks!

File: alv-portal-ui/src/app/competence-catalog/ch-fiches/ch-fiches-overview/ch-fiches-overview.component.ts (hunk 2 above)
Comment (cjpsycos) [reply alv-portal-ui/src/app/competence-catalog/ch-fiches/ch-fiches-overview/ch-fiches-overview.component.ts]: @ganqqwerty read my comment ;)

File: alv-portal-ui/src/app/shared/backend-services/reference-service/occupation-label.repository.ts [hunk 3]
```diff
@@ -51,12 +51,13 @@ export class OccupationLabelRepository {
    *
    * @param query the query/term
    * @param types of [x28', 'sbn3', 'sbn5'
+   * @param alternativeSearchUrl todo this is a temporary hack that is needed now when we have multiple reference services. This parameter needs to be removed when backend stabilises
    */
-  suggestOccupations(query: string, types: OccupationTypes[]): Observable<OccupationLabelAutocomplete> {
+  suggestOccupations(query: string, types: OccupationTypes[], alternativeSearchUrl?: string): Observable<OccupationLabelAutocomplete> {
```
Comment (birom) [inline alv-portal-ui/src/app/shared/backend-services/reference-service/occupation-label.repository.ts]: Hi @ganqqwerty,

I have started to do something similar in my [feature branch](https://github.com/alv-ch/alv-portal/blob/9e8c467295ab0fc86986e1be362746db95028535/alv-portal-ui/src/app/shared/backend-services/reference-service/occupation-label.repository.ts). We also need something similar for a POC.
Could you please have a look at it? I think, it makes sense if the ``OccupationLabelRepository`` controls also the alternative URL and the client of the repository just selects the version.

Thx.

**Comment by emina-str** [comment]
When I search the ChFiche by Beruf and enter a "blue" Occupation, I get this error:
image

> JSON parse error: Cannot deserialize value of type `ch.admin.seco.alv.service.competencecatalog.application.chfiche.ChFicheSearchRequest$OccupationCodeType` from String "CHISCO3": value not one of declared Enum instance names: [CHISCO5, BFS]; nested exception is com.fasterxml.jackson.databind.exc.InvalidFormatException: Cannot deserialize value of type `ch.admin.seco.alv.service.competencecatalog.application.chfiche.ChFicheSearchRequest$OccupationCodeType` from String "CHISCO3": value not one of declared Enum instance names: [CHISCO5, BFS]↵ at [Source: (PushbackInputStream); line: 1, column: 40] (through reference chain: ch.admin.seco.alv.service.competencecatalog.application.chfiche.ChFicheSearchRequest["occupationCodes"]->java.util.HashSet[0]->ch.admin.seco.alv.service.competencecatalog.application.chfiche.ChFicheSearchRequest$SearchOccupationCode["type"])

**Pull request merged**